src/reanimated/ReanimatedScreen.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { PropsWithChildren } from 'react';
+import { Platform, findNodeHandle, View } from 'react-native';
+import {
+  Screen,
+  ScreenProps,
+  ScreenContext,
+  TransitionProgressEventType,
+} from 'react-native-screens';
+
+// @ts-ignore file to be used only if `react-native-reanimated` available in the project
+import Animated, { makeMutable } from 'react-native-reanimated';
+// @ts-ignore file to be used only if `react-native-reanimated` available in the project, also types are not exported
+import WorkletEventHandler from 'react-native-reanimated/src/reanimated2/WorkletEventHandler';
+import ReanimatedTransitionProgressContext from './ReanimatedTransitionProgressContext';
+
+const AnimatedScreen = Animated.createAnimatedComponent(
+  (Screen as unknown) as React.ComponentClass
+);
+
+class ReanimatedScreen extends React.Component<ScreenProps> {
+  private ref: React.ElementRef<typeof View> | null = null;
+  private tag: number | null = null;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private eventHandler: WorkletEventHandler | null = null;
+  private progress = makeMutable(0);
+  private closing = makeMutable(0);
+  private goingForward = makeMutable(0);
+
+  setNativeProps(props: ScreenProps): void {
+    this.ref?.setNativeProps(props);
+  }
+
+  setRef = (ref: React.ElementRef<typeof View> | null): void => {
+    this.tag = findNodeHandle(ref);
+    this.ref = ref;
+    this.props.onComponentRef?.(ref);
+    if (this.eventHandler == null && this.tag != null) {
+      const progressRef = this.progress;
+      const closingRef = this.closing;
+      const goingForwardRef = this.goingForward;
+      this.eventHandler = new WorkletEventHandler(
```

Comment:
now that this component is extracted maybe we can use useEvent hook from reanimated instead of doing it this way?